### PR TITLE
Correct obstacle deletion

### PR DESCRIPTION
**Pull request by marcelkuipers**
Obstacles now get deleted when they are too far behind the commander.

### CODE REVIEW

File: ContextDesktop/src/main/java/com/github/migi_1/Context/obstacle/ObstacleSpawner.java [hunk 1]
```diff
@@ -42,6 +42,16 @@ public ObstacleSpawner(Commander commander) {
...
+
+        //call removeDamageDealer when an obstacle is too far away
+        ArrayList<Obstacle> deleteList = new ArrayList<Obstacle>();
+        for (Obstacle obs : obstacleList) {
+            if ((obs.getModel().getLocalTranslation().x - commander.getModel().getLocalTranslation().x) > 200) {
+                deleteList.add(obs);
+
+            }
+        }
+        obstacleList.removeAll(deleteList);
```
Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/obstacle/ObstacleSpawner.java:54]: Why is this happening the getObstacles() method though? This method seems to do much more than just getting the obstacles.

Comment (marcelkuipers) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/obstacle/ObstacleSpawner.java:54]: This method return the most up to date list of obstacles. Do you want me to rename or change it?

Comment (LMiljak) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/obstacle/ObstacleSpawner.java:54]: I would simply rename the method to something like updateObstacles() (the easy solution).
The other solution is to split it up in three methods: deleteObstacles, createObstacles and getObstacles.

Comment (marcelkuipers) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/obstacle/ObstacleSpawner.java:54]: I think that would be better too. Fixed it now

**Comment by LMiljak** [comment]
:+1: I approve

**Comment by ThatFlint** [comment]
:+1: so do I

**Pull request merged**